test/serializers/read_attribute_for_serialization_test.rb

```diff
@@ -7,20 +7,28 @@ class ReadAttributeForSerializationTest < ActiveSupport::TestCase
       class Parent < ActiveModelSerializers::Model
         attr_accessor :id
       end
+
       class Child < Parent
         attr_accessor :name
       end
+
       class ParentSerializer < ActiveModel::Serializer
         attributes :$id
 
         define_method(:$id) do
           object.id
         end
       end
+
       class ChildSerializer < ParentSerializer
         attributes :name
       end
 
+      def test_for_serializer_initialized_with_nil_object
+        child = ChildSerializer.new(nil)
+        assert_equal nil, child.read_attribute_for_serialization(:name)
```

Comment:
The problem with "silently failing" is that it makes misbehaving code go unnoticed and debugging very hard. When stuff fails, you know there's an issue with your code, and the sooner it fails, the easier it is to fix it.